src/openrct2/rct12/SawyerEncoding.cpp

```diff
@@ -24,6 +26,47 @@ extern "C"
 
 namespace SawyerEncoding
 {
+    void ReadChunk(void * dst, size_t expectedSize, IStream * stream)
+    {
+        if (!TryReadChunk(dst, expectedSize, stream))
+        {
+            throw IOException("Invalid or incorrect chunk size.");
+        }
+    }
+
+    void ReadChunkTolerant(void * dst, size_t expectedSize, IStream * stream)
+    {
+        uint64 originalPosition = stream->GetPosition();
+
+        auto header = stream->ReadValue<sawyercoding_chunk_header>();
+        switch (header.encoding) {
+        case CHUNK_ENCODING_NONE:
+        case CHUNK_ENCODING_RLE:
+        case CHUNK_ENCODING_RLECOMPRESSED:
+        case CHUNK_ENCODING_ROTATE:
+        {
+            std::unique_ptr<uint8> compressedData = std::unique_ptr<uint8>(Memory::Allocate<uint8>(header.length));
+            if (stream->TryRead(compressedData.get(), header.length) != header.length)
+            {
+                throw IOException("Corrupt chunk size.");
+            }
+
+            // Allow 16MiB for chunk data
+            size_t bufferSize = 16 * 1024 * 1024;
+            std::unique_ptr<uint8> buffer = std::unique_ptr<uint8>(Memory::Allocate<uint8>(bufferSize));
```

Comment:
Your `unique_ptr`s are wrong. You construct them with `malloc`, but delete with `operator delete` via `default_deleter`. Either provide proper deleter or allocate it with `operator new`.